src/v2/conversions.rs

```diff
@@ -914,19 +947,51 @@ impl TryFrom<AccountInfo> for super::types::AccountInfo {
             index,
             stake,
             address,
+            cooldowns,
+            available_balance,
         } = value;
         let account_nonce = sequence_number.require()?.into();
         let account_amount = amount.require()?.into();
-        let account_release_schedule = schedule.require()?.try_into()?;
+        let account_release_schedule: AccountReleaseSchedule = schedule.require()?.try_into()?;
         let account_threshold = threshold.require()?.try_into()?;
         let account_encrypted_amount = encrypted_balance.require()?.try_into()?;
         let account_encryption_key = encryption_key.require()?.try_into()?;
         let account_index = index.require()?.into();
-        let account_stake = match stake {
+        let account_stake: Option<super::types::AccountStakingInfo> = match stake {
             Some(stake) => Some(stake.try_into()?),
             None => None,
         };
         let account_address = address.require()?.try_into()?;
+        let mut cds: Vec<super::types::Cooldown> = Vec::with_capacity(cooldowns.len());
+        for cooldown in cooldowns.into_iter() {
+            cds.push(cooldown.try_into()?)
+        }
+        let cooldowns = cds;
+
+        // The available balance is only provided as convenience and in case the
+        // calculation of it changes in the future. If the available balance is
+        // not present, we calculate it manually instead. Note that changes in the
+        // calculation method may make this calculation invalid.
```

Comment:
If this field is missing, then it should mean that the node is pre version 7. In which case this calculation method covers how the available balance is calculated in all protocol versions supported by the node.

Reply:
Alright - how does that work in practice? I mean one node that uses protocol 7 and another that is pre version 7 will disagree on the available balance - how is it decided who is right? Is it just based on where the majority of stake is (i.e. if we assume most stake is on nodes that use protocol 7, then that is the truth)?

Reply:
According to the gRPC api we have, it should be present always otherwise we should mark this as optional for our own sanity.
I don't think we need the rust sdk to support every old version of the node software, so if node v7 always provide this information, we should be fine without this fallback calculation.

Reply:
Alright!

>According to the gRPC api (proto files) we have, it should be present always, if this is not the case we should mark this as optional for our own sanity.

Well, protobuf doesn't have any notion of required fields, everything is optional. The `optional` attribute just lets you tell the difference between an unset field (which will be decoded as the default value) and a field deliberately set to the default value - that's all it does. I think the only way we can signal that something is required is in application code or in documentation.